app/src/main/java/com/example/hlupico/mememaker/NewMemeActivity.java

```diff
@@ -1,31 +1,167 @@
 package com.example.hlupico.mememaker;
 
+import android.Manifest;
 import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.graphics.Bitmap;
+import android.net.Uri;
 import android.os.Bundle;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.FileProvider;
 import android.support.v7.app.AppCompatActivity;
+import android.text.format.DateFormat;
 import android.view.MenuItem;
 import android.view.View;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
 
 /**
  * Created by melder on 11/17/17.
  */
 
 public class NewMemeActivity extends AppCompatActivity {
+
+    private static final int REQUEST_CODE_CAMERA = 55;
+    private static final int REQUEST_CODE_GALLERY = 66;
+    private static final int REQUEST_CODE_TAKE_PHOTO = 77;
+    private static final int REQUEST_CODE_CHOOSE_PHOTO = 88;
+    private static final String[] CAMERA_PERMISSION = {Manifest.permission.CAMERA};
+    private static final String[] GALLERY_PERMISSION = {Manifest.permission.READ_EXTERNAL_STORAGE};
+    private Bitmap memeImage;
+    private Uri imageUri;
+    private Uri tempImageUri;
+    private EditText topText;
+    private EditText bottomText;
+
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_new_meme);
         setTitle("Make New Meme");
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        topText = (EditText)findViewById(R.id.top_text_input);
+        bottomText = (EditText)findViewById(R.id.bottom_text_input);
 
         findViewById(R.id.preview_button).setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 Intent intent = new Intent(NewMemeActivity.this, PreviewMemeActivity.class);
```

Comment:
Can you add a check to confirm there was an image selected and either top or bottom text before creating the Intent/starting the the PreviewMemeActivity? This is to guard against an NPE if the `Preview` button is selected before any of the fields have values.